zcash_client_backend/src/proposal.rs

```diff
@@ -48,6 +48,10 @@ pub enum ProposalError {
     PaymentPoolsMismatch,
     /// The proposal tried to spend a change output. Mark the `ChangeValue` as ephemeral if this is intended.
     SpendsChange(StepOutput),
+    /// The proposal attempts to send a memo to a transparent recipient.
+    SendsMemoToTransparentRecipient,
+    /// The proposal results in an invalid payment request.
+    ResultingPaymentInvalid(Zip321Error),
```

Comment:
```suggestion
    Zip321(Zip321Error),
```

We typically attempt to name error wrappers more like this.